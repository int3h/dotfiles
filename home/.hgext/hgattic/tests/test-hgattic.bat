@echo off

rem setup hgattic in a temp test directory
setlocal
cd ..
cd > tmpfile
set /p atticdir= < tmpfile
erase tmpfile
cd tests
mkdir test
cd test
call hg init
cd .hg
echo [extensions] >tmp
echo hgattic=%atticdir%\attic.py >>tmp
move tmp hgrc
cd ..

rem replace here with test content
call hg help hgattic
rem end test content

rem cleanup
cd ..
rmdir /S /Q test
endlocal
