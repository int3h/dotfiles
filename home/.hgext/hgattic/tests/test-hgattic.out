 hgattic extension - manage uncommitted changes with a set of named patches

This extension lets you create patches from uncommited changes using its
'shelve' command. Shelved changes are unapplied from the working copy and
stored as patches in the .hg/attic directory.

They can be later restored using the 'unshelve' command, which merges the
changes back into the working copy.

This allows task switching between many patches in a single repository.

As applied patches are just changes in the working copy they are not part of
the project history but can, of course, be commited if desired.

Having all known patches in the .hg/attic directory allows you to easily share
patch sets between repositories and even control version them.

Common tasks (use 'hg help command' for more details):

attic-shelve (shelve):
    store the current working copy changes in a patch in the attic and prepare
    to work on something else unapplying those changes
attic-display (attic, ls):
    list the patches in the attic
attic-unshelve (unshelve):
    activate a patch to work on and merge its changes into the working copy

list of commands:

 attic-display    lists the available patches in the attic
 attic-shelve     move changes from working copy to the attic
 attic-unshelve   applies a patch from the attic to the working copy

use "hg -v help hgattic" to show aliases and global options
